### ISSUE TITLE
extrudeRotate is broken when input polygon is clockwise

### ISSUE THREAD

**Issue by paulftw** [jscad/OpenJSCAD.org#1012]
## Steps to Reproduce the Problem
Paste below code into `openjscad.xyz` (two triangles are rotateExtruded, but in the first case the vertices  are counterclockwise).

You'll see rendering artifacts on the bottom extrusion. Boolean operations with an "inside out" extrusion also give weird results.

```
const {
  booleans,
  extrusions,
  geometries,
  maths,
  primitives,
  transforms,
} = require('@jscad/modeling')

const { circle } = primitives
const { geom2 } = geometries
const { mat4 } = maths

const main = () => {
  return [
    transforms.translate(
      [0, 0, 20],
      extrusions.extrudeRotate(
        { segments: 32, startAngle: 0, angle: Math.PI * 0.5, overflow: 'cap' },
        primitives.polygon({
          points: [
            [-10, -0],
            [-10, -10],
            [-15, -5],
          ],
        }),
      ),
    ),

    extrusions.extrudeRotate(
      { segments: 32, startAngle: 0, angle: Math.PI * 0.5, overflow: 'cap' },
      primitives.polygon({
        points: [
          [10, 0],
          [10, 10],
          [15, 5],
        ],
      }),
    ),
  ]
}

module.exports = {main}
```

**Comment by hrgdavor** [comment]
My opinion is that library should not defensively call all sorts of fixes for the input.
This would allow code like above  to work, but would punish with performance degradation those that provide good input geometry.

I would be against input cleanup, and in favor of making clear in docs that input must be ccw.

**Comment by platypii** [comment]
This is expected behavior. Most computational geometry assumes that polygons are oriented in a counter-clockwise direction. This is even more important in 3D where a polygon can face in or out. However even in 2D, counterclockwise is considered a solid, and clockwise is used to represent holes, or negative space. That is why extruding a negative space does not totally make sense.

I am against "automatic flipping" of inverted polygons. It might be better to throw an exception, rather than produce a flawed extrusion. But overall this is expected. More docs are a good idea though. (for the record, I ran into the same confusion #808)

**Comment by z3dev** [comment]
@paulftw right or wrong, there might be cases where someone wants to create an inverted extrusion. i know that some users are using inverted 3D shapes as 'holes' when using the booleans.

i also looked at the user guide. it doesn't mention anything about the 'handed-ness' of shapes, so probably a document change would be good.

**Comment by paulftw** [comment]
A mention in the docs would be nice. An optional flag (which most people will just switch to true) or at least a utility function to detect directionality would be even better.

Supporting all possible use cases and prioritizing the most common ones aren't mutually exclusive goals. And these goals have been combined in other places in the JSCAD API.
(e.g. some people want to set alpha channel explicitly, but those who pass a three element vector [are forgiven](https://github.com/jscad/OpenJSCAD.org/blob/a63ddb02d7640af85276192dd8d1ef8f6bfc8818/packages/modeling/src/colors/colorize.js#L48); there's a `cube` primitive, which is "completely unnecessary"  since `cuboid` can easily do its job; etc.).

**Comment by z3dev** [comment]
@paulftw yeah. we could follow suit, and provide an 'orientation' parameter.

`
const myshape = polyhedron({points, faces, orientation: 'inward'})
const myshape = polygon({points, orientation: 'clockwise'})
`

**Comment by z3dev** [comment]
@paulftw the orientation option has been added. Can this issue be closed?

**Comment by z3dev** [comment]
Closing as the orientation option is available.
